lib/authority.js

```diff
@@ -214,12 +221,19 @@ Authority.prototype._validateViaInstanceDiscovery = function(callback) {
  * @param {Authority.GetOauthEndpointsCallback}  callback  The callback function.
  */
 Authority.prototype._getOAuthEndpoints = function(tenantDiscoveryEndpoint, callback) {
-  if (this._tokenEndpoint) {
+  if (this._tokenEndpoint && this._deviceCodeEndpoint) {
     callback();
     return;
   } else {
     // fallback to the well known token endpoint path.
-    this._tokenEndpoint = url.format(this._url) + AADConstants.TOKEN_ENDPOINT_PATH;
+    if (!this._tokenEndpoint){
+       this._tokenEndpoint = url.format(this._url) + AADConstants.TOKEN_ENDPOINT_PATH;
+    }
+
+    if (!this._deviceCodeEndpoint){
+       this._deviceCodeEndpoint = url.format(this._url) + AADConstants.DEVICE_ENDPOINT_PATH;
```

Comment:
`url.format` already adds a `/` at the end and the AADConstants paths also add a `/`.

`url.format("http://localhost:9292") + AADConstants.DEVICE_ENDPOINT_PATH; = "http://localhost:9292//oauth2/devicecode"`

Most browsers will ignore the double slash, but as far as I know they don't have to. We should probably fix this for the other paths as well.